configure.ac

```diff
@@ -105,6 +107,16 @@ AC_CHECK_LIB(dl, dlopen, [],
 ])
 fi
 
+dnl handle picky option
+AC_ARG_ENABLE([picky],
+    [AC_HELP_STRING([--enable-picky],
+                    [Enable developer-level compiler pickyness when building @<:@default=no@:>@])])
+AS_IF([test x"$enable_picky" == x"yes" -a x"$GCC" = x"yes"],
```

Comment:
Don't use `==` with `test`; some flavors of `test` will do unexpected things.

But you get doubleplusgood points for converting to `AS_IF`.  👏